src/integrationTest/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTaskTest.java

```diff
@@ -50,7 +50,7 @@ public void uploads_file_to_sftp_and_sets_letter_status_to_uploaded() throws Exc
             // Ensure the letter is marked as uploaded in the database.
             // Clear the JPA cache to force a read.
             entityManager.clear();
-            Letter l = repository.findById(id).get();
+            Letter l = repository.findOptionalById(id).get();
```

Comment:
Will this compile?

Reply:
integration tests pass - application loads context successfully. the tests show it works

link to documentation says it is a way to avoid unnecessary exception. there would still be an error in case `id` is passed as null

Reply:
Oh, obviously. Missed the `get` at the end. And the fact that it was Optional before, as well. All fine.